test/rotation_matrix_test.cpp

```diff
@@ -0,0 +1,136 @@
+
+  Eigen::Matrix3f input_matrix;
+  input_matrix << -1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, -1.0;
+
+  double x_dim, y_dim;
+  Eigen::Matrix3f actual_matrix = StandardizeBoxOrientation(
+      input_matrix, kShortSide, kLongSide, &x_dim, &y_dim);
+
+  ASSERT_TRUE(expected_matrix.isApprox(actual_matrix, 0.0001));
+}
+
+TEST(TestStandardizeBoxOrientation, IdentityMatrixRotate45DegreesAroundZAxis) {
+  Eigen::Matrix3f expected_matrix;
+  expected_matrix << 0.70711, -0.70711, 0.0, 0.70711, 0.70711, 0.0, 0.0, 0.0,
+      1.0;
+
+  Eigen::Matrix3f input_matrix;
+  input_matrix << 0.70711, -0.70711, 0.0, 0.70711, 0.70711, 0.0, 0.0, 0.0, 1.0;
+
+  double x_dim, y_dim;
+  Eigen::Matrix3f actual_matrix = StandardizeBoxOrientation(
+      input_matrix, kShortSide, kLongSide, &x_dim, &y_dim);
+
+  ASSERT_TRUE(expected_matrix.isApprox(actual_matrix, 0.0001));
+}
+
+TEST(TestStandardizeBoxOrientation, IdentityMatrixRotate135DegreesAroundZAxis) {
+  Eigen::Matrix3f expected_matrix;
+  expected_matrix << 0.70711, -0.70711, 0.0, 0.70711, 0.70711, 0.0, 0.0, 0.0,
+      1.0;
+
+  Eigen::Matrix3f input_matrix;
+  input_matrix << -0.70711, 0.70711, 0.0, -0.70711, 0. - 70711, 0.0, 0.0, 0.0,
+      1.0;
+
+  double x_dim, y_dim;
+  Eigen::Matrix3f actual_matrix = StandardizeBoxOrientation(
+      input_matrix, kShortSide, kLongSide, &x_dim, &y_dim);
+
+  ASSERT_TRUE(expected_matrix.isApprox(actual_matrix, 0.0001));
+}
+
+TEST(TestStandardizeBoxOrientation, TiltedMatrix) {
+  Eigen::Matrix3f expected_matrix;
+  expected_matrix << 0.99980, 0.01732, 0.01, -0.02, 0.86580, 0.5, 0.0, -0.5001,
+      0.86597;
+
+  Eigen::Matrix3f input_matrix;
+  input_matrix << 0.99980, 0.01732, 0.01, -0.02, 0.86580, 0.5, 0.0, -0.5001,
+      0.86597;
+
+  double x_dim, y_dim;
+  Eigen::Matrix3f actual_matrix = StandardizeBoxOrientation(
+      input_matrix, kShortSide, kLongSide, &x_dim, &y_dim);
+
+  ASSERT_TRUE(expected_matrix.isApprox(actual_matrix, 0.0001));
+}
+
+TEST(TestStandardizeBoxOrientation, TiltedMatrixRotate180DegreesAroundYAxis) {
+  Eigen::Matrix3f expected_matrix;
+  expected_matrix << 0.99980, -0.01732, -0.01, -0.02, -0.86580, -0.5, 0.0,
+      0.5001, -0.86597;
+
+  Eigen::Matrix3f input_matrix;
+  input_matrix << -0.99980, 0.01732, -0.01, 0.02, 0.86580, -0.5, 0.0, -0.5001,
+      -0.86597;
+
+  double x_dim, y_dim;
+  Eigen::Matrix3f actual_matrix = StandardizeBoxOrientation(
+      input_matrix, kShortSide, kLongSide, &x_dim, &y_dim);
+
+  ASSERT_TRUE(expected_matrix.isApprox(actual_matrix, 0.0001));
+}
+
+TEST(TestStandardizeBoxOrientation, IdentityMatrixWithXLongSideYShortSdie) {
+  Eigen::Matrix3f expected_matrix;
+  expected_matrix << 0.0, -1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0;
+
+  Eigen::Matrix3f input_matrix;
+  input_matrix << 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0;
+
+  double x_dim, y_dim;
+  Eigen::Matrix3f actual_matrix = StandardizeBoxOrientation(
+      input_matrix, kLongSide, kShortSide, &x_dim, &y_dim);
```

Comment:
You should also add

```cpp
EXPECT_EQ(kShortSide, x_dim);
EXPECT_EQ(kLongSide, y_dim);
```

This should be in all the test cases, but is especially important to test here.

Reply:
Also, note the difference between ASSERT and EXPECT in googletest. ASSERT will fail the test as soon as the first ASSERT fails. If an EXPECT fails, then the test will fail, but it will keep running.

Generally, EXPECT is preferred, since that way you can get more information at once:

```cpp
ASSERT_TRUE(a); // fails, test is over!
ASSERT_TRUE(b); // unknown!
ASSERT_TRUE(c); // unknown!
```

```cpp
EXPECT_TRUE(a); // fails
EXPECT_TRUE(b); // passes
EXPECT_TRUE(c); // fails
```